src/server/start_service.py

```diff
@@ -29,37 +34,131 @@ def create_periodic_introducer_poll_task(
     server,
     peer_info,
     global_connections,
-    introducer_connect_interval,
-    target_peer_count,
+    peer_connect_interval,
+    target_outbound_connections,
 ):
     """
 
     Start a background task connecting periodically to the introducer and
     requesting the peer list.
     """
 
+    def poisson_next_send(now, avg_interval_seconds):
+        return now + (
+            math.log(
```

Comment:
Maybe a comment to explain what the hell this is really trying to do (could just be "poisson distribution, because bitcoin does it, with the following parameters: ..." and link to Wikipedia article about the dist).